### ISSUE TITLE
Deadlock when using dill and an exception is thrown

### ISSUE THREAD

**Issue by derHeinzer** [Slimmer-AI/mpire#56]
When using dill and an exception is thrown in one of the worker processes the main process deadlocks.

reproducable example:
```
from mpire import WorkerPool

# Create some fake data. It should raise when x=10, y=52, z=0
data = [(x, y, z) for x, y, z in zip(range(0, 100), range(42, 142), range(10, -90, -1))]
with WorkerPool(n_jobs=5, use_dill=True) as pool:
    for res in pool.imap(lambda x, y, z: x*y/z, data):
        print(res)

```

Wwhen setting use_dill = False it works like a charm. Unfortunaeley in my case I have to use dill to pickle lambda functions.

**Comment by derHeinzer** [comment]
It seems, the subprocess that puts the exception to the exception_queue dies BEFORE the exception is flushed to the pipe. According to the docs this is NOT the desired default behaviour.

A solution that works in my case would be to close the exception queue after putting the exception to the queue by adding 

''''
self.worker_comms._exception_queue.close()
'''

in line 416 of worker.py


https://docs.python.org/3.4/library/multiprocessing.html?highlight=process
Warning As mentioned above, if a child process has put items on a queue (and it has not used [JoinableQueue.cancel_join_thread](https://docs.python.org/3.4/library/multiprocessing.html?highlight=process#multiprocessing.Queue.cancel_join_thread)), then that process will not terminate until all buffered items have been flushed to the pipe.
This means that if you try joining that process you may get a deadlock unless you are sure that all items which have been put on the queue have been consumed. Similarly, if the child process is non-daemonic then the parent process may hang on exit when it tries to join all its non-daemonic children.

Note that a queue created using a manager does not have this issue. See [Programming guidelines](https://docs.python.org/3.4/library/multiprocessing.html?highlight=process#multiprocessing-programming).

**Comment by sybrenjansen** [comment]
Thnx for the bug report and apologies for the late reply (holiday season).

I can confirm it deadlocks here on my end as well. I will look into it soon

**Comment by sybrenjansen** [comment]
It's currently fixed in the `exceptions` branch (it's volatile though). I will attend to a few more bug reports and then make a new release.

**Comment by sybrenjansen** [comment]
Fixed in v2.6.0
